Description: Allow override of SECRET_KEY
 As dashboard plugins are importing openstack_dashboard/test/settings.py, the
 import fails with secret_key.generate_or_read_from_file() attempting to write
 in /usr/lib/python2.7/dist-packages/openstack_dashboard/test.
 .
 With this patch, if plugins define SECRET_KEY first (in a writeable path),
 then Horizon's openstack_dashboard/test/settings.py wont attempt to write it
 in /usr.
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Last-Update: 2016-09-29

--- horizon-10.0.0~rc2.orig/openstack_dashboard/test/settings.py
+++ horizon-10.0.0~rc2/openstack_dashboard/test/settings.py
@@ -33,8 +33,12 @@ STATIC_ROOT = os.path.abspath(os.path.jo
 STATIC_URL = '/static/'
 WEBROOT = '/'
 
-SECRET_KEY = secret_key.generate_or_read_from_file(
-    os.path.join(TEST_DIR, '.secret_key_store'))
+try:
+    SECRET_KEY
+except:
+    SECRET_KEY = secret_key.generate_or_read_from_file(
+        os.path.join(TEST_DIR, '.secret_key_store'))
+
 ROOT_URLCONF = 'openstack_dashboard.test.urls'
 
 TEMPLATES[0]['DIRS'] = [
